libsplinter/src/biome/rest_api/actix/user.rs

```diff
@@ -314,27 +318,43 @@ fn add_delete_user_method(
             }
         };
 
-        Box::new(match user_store.remove_user(&user_id) {
-            Ok(()) => HttpResponse::Ok()
-                .json(json!({ "message": "User deleted sucessfully" }))
-                .into_future(),
-            Err(err) => match err {
-                UserStoreError::NotFoundError(msg) => {
-                    debug!("User not found: {}", msg);
+        match user_store.remove_user(&user_id) {
+            Ok(()) => (),
+            Err(UserStoreError::NotFoundError(msg)) => {
+                debug!("User not found: {}", msg);
+                return Box::new(
                     HttpResponse::NotFound()
                         .json(ErrorResponse::not_found(&format!(
                             "User ID not found: {}",
                             &user_id
                         )))
-                        .into_future()
-                }
-                _ => {
-                    error!("Failed to delete user in database {}", err);
+                        .into_future(),
+                );
+            }
+            Err(err) => {
+                error!("Failed to delete user in database {}", err);
+                return Box::new(
                     HttpResponse::InternalServerError()
                         .json(ErrorResponse::internal_error())
-                        .into_future()
-                }
-            },
-        })
+                        .into_future(),
+                );
+            }
+        };
+
+        match credentials_store.remove_credentials(&user_id) {
+            Ok(()) => Box::new(
+                HttpResponse::Ok()
+                    .json(json!({ "message": "User deleted sucessfully" }))
+                    .into_future(),
+            ),
+            Err(err) => {
+                error!("Failed to delete user in database {}", err);
```

Comment:
This error message should probably be different than the one for the users store so they're easily distinguishable in the logs.

Reply:
Changed to "Failed to delete credentials in database"